### PR TITLE
Change the way update_localdb() works

### LINKED ISSUE #95: Error: Could Not Rename: ... to ...
## Reproduction Rate
On each `tldr <command>` run without local database

## Steps to Reproduce
1. `rm -rf ~/.tldrc`
2. `tldr tldr`

## Result
Couple seconds of delay (downloading the tldr database), then the error message `Error: Could Not Rename: /tmp/tldrXXXXXX/tldr-main to /home/<user>/.tldrc/tldr/`, then the actual tldr page.
`strace` shows that the [rename](https://github.com/tldr-pages/tldr-c-client/blob/903b38d5fa2b97ee8b72e151efbe6b797618059d/src/local.c#L213) call fails with EXDEV (Invalid cross-device link).

## Expected Result
Normal output witout error

## Additional Information
```
$ tldr --version
tldr v1.5.0 (v1.5.0-3-g903b38d)
Copyright (C) 2016 Arvid Gerstmann
Source available at https://github.com/tldr-pages/tldr-c-client
```
Arch Linux, tldr installed from [tldr-git](https://aur.archlinux.org/packages/tldr-git)

### PR DESCRIPTION
**Pull request by Ar7eniyan**
## What does it do?
Makes the update of a local database not use `/tmp/tldrXXXXXX` temp directory, but download the update archive to a temporary place inside a tldr home directory (currently `~/.tldrc/tmp/`), which omits painful cross-FS file manipulation.

## Why the change?
Makes the local database work on systems where `/tmp/` and `$HOME` are on different partitions. Resolves #95.

## How can this be tested?
Remove tldr home folder (`~/.tldrc`) and run `tldr -u`

## Where to start code review?
https://github.com/tldr-pages/tldr-c-client/pull/100/commits/c8424d422b9229b7c05b1c061547331ca5cde92a

## Relevant tickets?
#95 

## Questions?
C isn't my primary language, so any corrections are welcome.

### CODE REVIEW

**Comment by Ar7eniyan** [comment]
Is there any maintainer to review the PR?

**Review by 4G3NT — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]

File: src/local.c [hunk 1]
```diff
@@ -140,87 +140,102 @@ int
...
+    if (sstrncat(tldr_home_db, &outlen, STRBUFSIZ, tldr_home, strlen(tldr_home))) {
         return 1;
     }
-
-    if (sstrncat(outhome, &outlen, STRBUFSIZ, TLDR_DIR, TLDR_DIR_LEN))
+    if (sstrncat(tldr_home_db, &outlen, STRBUFSIZ, TLDR_DIR, TLDR_HOME_LEN)) {
         return 1;
-    if (sstrncat(outhome, &outlen, STRBUFSIZ, "/", 1))
+    }
+    if (sstrncat(tldr_home_db, &outlen, STRBUFSIZ, "/", 1)) {
```
Comment (SethFalco) [inline src/local.c:219]: I don't think the trailing slash is necessary. Can you confirm?
If not, then it should be safe to remove these 3 lines.

I've tested your branch without them, and it appears to work fine.

Maybe it'd cause issues if we had a file named `tldr` in the same directory, but that's not the case. Given that the state of this directory is managed by us, I don't think it's a concern.

File: src/local.c [hunk 2]
```diff
@@ -140,87 +140,102 @@ int
...
 {
     struct stat s;
-    char tmp[STRBUFSIZ];
-    char outpath[STRBUFSIZ];
-    char outfile[STRBUFSIZ];
-    char outhome[STRBUFSIZ];
+    char tldr_home[STRBUFSIZ];         /* $HOME/TLDR_HOME                      */
+    char tldr_home_db[STRBUFSIZ];      /* $HOME/TLDR_HOME/TLDR_DIR             */
+    char temp_dir[STRBUFSIZ];          /* $HOME/TLDR_HOME/TMP_DIR              */
+    char update_zip_file[STRBUFSIZ];   /* $HOME/TLDR_HOME/TMP_DIR/TMP_FILE     */
```
Comment (SethFalco) [inline src/local.c:146]:
```suggestion
    char zip_archive_path[STRBUFSIZ];   /* $HOME/TLDR_HOME/TMP_DIR/TMP_FILE     */
```

Think we could rename this variable to `zip_archive_path`?

1. I think `path` is more appropriate than `file` when handling paths.
2. `update_zip` isn't a clear name imo.

File: src/local.c [hunk 3]
```diff
@@ -140,87 +140,102 @@ int
...
+    char update_contents[STRBUFSIZ];   /* $HOME/TLDR_HOME/TMP_DIR/TLDR_ZIP_DIR */
```
Comment (SethFalco) [inline src/local.c:147]:
```suggestion
    char extracted_contents[STRBUFSIZ];   /* $HOME/TLDR_HOME/TMP_DIR/TLDR_ZIP_DIR */
```

Think we could rename this variable to `extracted_contents` or `zip_contents`?

**Review by SethFalco — COMMENTED** [review]
I'm no C expert, but this looks good to me overall. Also tested it and works flawlessly.

I've left a few comments, but all of them are subjective. Anyone is welcome to dispute them.

**Review by kbdharun — APPROVED** [review]

**Review by SethFalco — APPROVED** [review]

**Review by pixelcmtd — APPROVED** [review]

**Pull request merged**